src/containers/App.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-param-reassign */
+import React, { PureComponent } from 'react';
+import styled from 'styled-components';
+import { moviesData, filtersData } from '../data';
+import Header from '../components/Header';
+import MoviesResult from './MoviesResult';
+import Footer from '../components/Footer';
+import ErrorBoundary from './ErrorBounding';
+
+class App extends PureComponent {
+    // eslint-disable-next-line react/state-in-constructor
+    state = {
+        searchValue: '',
+        sortValue: 'date',
+        filters: filtersData,
+        all: true,
+        movies: moviesData
+    }
+
+    searchOnChange = (event) => {
+        this.setState({
+            searchValue: event.target.value
+        });
+    }
+
+    dynamicSearch = () => {
+        const { movies, searchValue } = this.state;
+
+        const searchMovies = searchValue
+            ? movies.filter(movie => movie.title.toLowerCase().includes(searchValue.toLowerCase()))
+            : moviesData;
+
+        return searchMovies;
+    }
+
+    handleSearch = () => {
+        const searchResult = this.dynamicSearch();
+        this.setState({ movies: searchResult });
+    };

```

Comment:
From my point of view method `dynamicSearch` is redundant. You can use just like that
```jsx
handleSearch = () => {
        const { movies, searchValue } = this.state;

        this.setState({
            movies: searchValues
                ? movies.filter(({ title }) => (title.toLowerCase().includes(searchValue.toLowerCase()))) // It's possible to create a helper to make more readable
                : this.state.defaultMoviesData, // If you use api you will make api call or save default data into separate field
        })
}
```